From 2f72968ec33e87185b36bb9c45450434de341fe6 Mon Sep 17 00:00:00 2001
From: sunchendong <sunchendong@xfusion.com>
Date: Fri, 26 Dec 2025 10:34:55 +0800
Subject: [PATCH] [xfusion]vLLM-ascend:The ears feature is compatible with
 Suffix Decoding.

Offering:AI Solution 25
CVE:
Reference:
Type:requirement
DTS/US:US20251111238969
reason:vLLM-ascend:The ears feature is compatible with Suffix Decoding.

Signed-off-by: sunchendong <sunchendong@xfusion.com>
---
 vllm_ascend/worker/model_runner_v1.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vllm_ascend/worker/model_runner_v1.py b/vllm_ascend/worker/model_runner_v1.py
index 7d4343e..e579896 100644
--- a/vllm_ascend/worker/model_runner_v1.py
+++ b/vllm_ascend/worker/model_runner_v1.py
@@ -416,7 +416,7 @@ class NPUModelRunner(GPUModelRunner):
                 # EARS (Entropy-Adaptive Rejection Sampling) configuration
                 ears_tolerance = envs_ascend.VLLM_EARS_TOLERANCE
 
-                if ears_tolerance > 0 and self.speculative_config.method == "eagle3":
+                if ears_tolerance > 0 and self.speculative_config.method in ["eagle3", "suffix"]:
                     self.rejection_sampler = EntropyAdaptiveRejectionSampler(
                         self.sampler,
                         base_tolerance=ears_tolerance,
-- 
2.43.0

